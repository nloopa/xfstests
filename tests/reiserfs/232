#! /bin/bash
# FSQA Test No. 232
#
# Run fsstress with quotas enabled and verify accounted quotas in the end
#
# Derived from test 231
#-----------------------------------------------------------------------
# Copyright (c) 2006 Silicon Graphics, Inc.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "rm -f $tmp.*; exit \$status" 0 1 2 3 15

# get standard environment, filters and checks
. ./common/rc
. ./common/filter
. ./common/quota

_filter_num()
{
	tee -a $seqres.full |\
	sed -e 's/[0-9][0-9]* inodes/I inodes/g' \
	    -e 's/[0-9][0-9]* paths/P paths/g' \
	    -e 's/seed = [0-9][0-9]*/seed = S/'
}

_fsstress()
{
	echo ""
	echo "Testing fsstress"
	echo ""

	out=$SCRATCH_MNT/fsstress.$$
	count=2000
	args=`_scale_fsstress_args -d $out -n $count -p 7`

	echo "fsstress $args" >> $seqres.full
	if ! $FSSTRESS_PROG $args | tee -a $seqres.full | _filter_num
	then
		echo "    fsstress $args returned $?"
		cat $tmp.out | tee -a $seqres.full
		status=1
	fi
}

# real QA test starts here
_supported_fs generic
_supported_os Linux
_require_scratch
_require_quota
_need_to_be_root

_scratch_mkfs >> $seqres.full 2>&1
_scratch_mount "-o usrquota,grpquota"
chmod 777 $SCRATCH_MNT
quotacheck -u -g $SCRATCH_MNT 2>/dev/null
quotaon -u -g $SCRATCH_MNT 2>/dev/null

if ! _fsstress; then
	umount $SCRATCH_DEV 2>/dev/null
	exit
fi

if ! _check_quota_usage; then
	umount $SCRATCH_DEV 2>/dev/null
	status=1
	exit
fi

umount $SCRATCH_DEV 2>/dev/null
status=0
exit
