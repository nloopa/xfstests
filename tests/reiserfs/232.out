QA output created by 232

Testing fsstress

seed = S
Comparing user usage
Comparing group usage
